core/src/main/java/io/grpc/transport/ServerTransportListener.java

```diff
@@ -48,4 +49,9 @@
    */
   ServerStreamListener streamCreated(ServerStream stream, String method,
       Metadata.Headers headers);
+
+  /**
+   * The transport completed shutting down. All resources have been released.
+   */
+  void transportTerminated();
```

Comment:
I had transportShutdown(), but it didn't prove useful so I removed it.

"close" has very weak semantics for something like this. It cannot mean both "shutdown" and "terminated" as the two words mean different things and I am depending on the details from each.